### PR TITLE
DDCE-3738 Update the CSP policy

### PR DESCRIPTION
**Pull request by mywyau**
Screenshot 2023-03-31 at 10 03 45

### CODE REVIEW

**Comment by mywyau** [comment]
Unit and ATs should still pass

Screenshot 2023-03-31 at 12 00 24

File: app/views/govuk_wrapper.scala.html [hunk 1]
```diff
@@ -33,10 +32,14 @@
...
-@(title: PageTitle, beforeContentHtml: Option[Html] = None, researchBannerUrl: Option[String] = None,
+@(
+    title: PageTitle,
+    beforeContentHtml: Option[Html] = None,
+    researchBannerUrl: Option[String] = None,
     signOut: Boolean = true,
     timeoutActive: Boolean = true,
-    backLink: Option[String]= None)(mainContent: Html)(implicit request: Request[_], messages: Messages, appConfig: config.AppConfig)
+    backLink: Option[String]= None,
+)(mainContent: Html)(implicit request: Request[_], requestHeader: RequestHeader, messages: Messages, appConfig: config.AppConfig)
```
Comment (frikit) [inline app/views/govuk_wrapper.scala.html:42]: please remove `requestHeader: RequestHeader` as not used anywhere

**Review by frikit — CHANGES_REQUESTED** [review]
- Please update dependencies

File: app/config/FeatureSwitching.scala [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  def isDisabled(featureSwitch: FeatureSwitch)(implicit config: AppConfig): Boolean =
+    !getValue(featureSwitch).toBoolean
+
+  def getValue(featureSwitch: FeatureSwitch)(implicit config: AppConfig): String =
+    getValue(featureSwitch.name)
+
+  def getValue(key: String)(implicit config: AppConfig): String =
+    sys.props.get(key).getOrElse(config.servicesConfig.getString(key))
+
+  def setValue(featureSwitch: FeatureSwitch, value: String): SystemProperties =
```
Comment (frikit) [inline app/config/FeatureSwitching.scala:40]: we never supported changing config values from code and I dont see a need here as well, please delete all `setValue` , `enable`, `disable`, `fsGuard`

File: app/config/FeatureSwitching.scala [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ * limitations under the License.
+ */
+
+package config
+
+import config.AppConfig
+
+import scala.sys.SystemProperties
+
+trait FeatureSwitching {
```
Comment (frikit) [inline app/config/FeatureSwitching.scala:23]: I would say this class is too much for current use case, you can just add a method inside your impl like `StudentLoanFlagFeature` and trigger that one, you can add all required info like type class in there and method can be generic in interface like getValue[A] and should be simple and good enough to use it

File: test/controllers/EmploymentDetailControllerSpec.scala [hunk 4]
```diff
@@ -35,205 +36,259 @@ import views.html.taxhistory.employment_detail
...
+        }
+      }
+
+    }
+
     "successfully load Employment details page" in new LocalSetup {
 
       val result: Future[Result] =
-        controller.getEmploymentDetails(employmentId.toString, taxYear)(fakeRequest.withSession("USER_NINO" -> nino))
+        c.getEmploymentDetails(employmentId.toString, taxYear)(fakeRequest.withSession("USER_NINO" -> nino))
```
Comment (frikit) [inline test/controllers/EmploymentDetailControllerSpec.scala:166]: revert back name here to be more descriptive

Comment (frikit) [reply test/controllers/EmploymentDetailControllerSpec.scala:166]: in this class

**Review by frikit — CHANGES_REQUESTED** [review]
- update all deps

**Review by frikit — CHANGES_REQUESTED** [review]
- update all deps

**Comment by mywyau** [comment]
Old Coverage

**Comment by mywyau** [comment]
**New code coverage**
Screenshot 2023-04-11 at 15 34 17

**Comment by mywyau** [comment]
- Reverted to remove feature switch class
- Updated dependencies 
- Improved test coverage, adding in more tests.

**Review by frikit — APPROVED** [review]

**Pull request merged**